src/exec/handle_io.c

```diff
@@ -18,27 +18,33 @@
 #include "ctx.h"
 #include "exec_internal.h"
 
-static void	handle_pipeline(t_pipeline_conf *conf)
+/*
+** next_read should be closed only if the process is forked
+*/
+static void	handle_pipeline(t_pipe_conf *conf, bool is_forked)
 {
-	if (conf->fd_in != STDIN_FILENO)
+	if (conf == NULL)
+		return ;
+	if (conf->prev_read != STDIN_FILENO)
 	{
-		if (dup2(conf->fd_in, STDIN_FILENO) == -1)
+		if (dup2(conf->prev_read, STDIN_FILENO) == -1)
 			print_error("dup2", strerror(errno));
-		else
-			close(conf->fd_in);
+		close(conf->prev_read);
 	}
-	if (conf->fd_out != STDOUT_FILENO)
+	if (conf->next_write != STDOUT_FILENO)
 	{
-		if (dup2(conf->fd_out, STDOUT_FILENO) == -1)
+		if (dup2(conf->next_write, STDOUT_FILENO) == -1)
 			print_error("dup2", strerror(errno));
-		else
-			close(conf->fd_out);
+		close(conf->next_write);
 	}
+	if (is_forked && conf->next_read != -1 && conf->next_read != STDIN_FILENO)
```

Comment:
is there actual case that `conf->next_read == STDIN_FILENO` ?
```suggestion
	if (is_forked && conf->next_read != -1)
```

Reply:
No... The former is better.